### ISSUE TITLE
DB/PreparedSQLPlaceholders: false negatives for namespaced and method calls to sprintf()

### ISSUE THREAD

**Issue by rodrigoprimo** [WordPress-Coding-Standards/WordPress-Coding-Standards#2720]
## Bug Description

The `WordPress.DB.PreparedSQLPlaceholders` sniff incorrectly treats namespaced function calls and method calls to `sprintf()` as if they were calls to the global `sprintf()` function, causing false negatives.

The sniff has special handling for the dynamic placeholder pattern using `sprintf()`. When it encounters a namespaced call like `MyNamespace\sprintf()` or a method call like `$obj->sprintf()`, it incorrectly applies this special handling instead of treating the call as an unknown function.

This issue only affects `sprintf()`. The detection of `implode()` and `array_fill()` already correctly filters out namespaced and method calls.

## Minimal Code Snippet

The issue happens when running this command:
```bash
phpcs --standard=WordPress --sniffs=WordPress.DB.PreparedSQLPlaceholders test.php
```

... over a file containing this code:

```php
<?php

$wpdb->prepare(
	MyNamespace\sprintf(
		'SELECT * FROM `table` WHERE `ID` IN (%s)',
		implode( ',', array_fill( 0, count( $ids ), '%s' ) )
	),
	$ids
);

$wpdb->prepare(
	$obj->sprintf(
		'SELECT * FROM `table` WHERE `ID` IN (%s)',
		implode( ',', array_fill( 0, count( $ids ), '%s' ) )
	),
	$ids
);
```

### Expected behavior:

Both lines should be flagged. `MyNamespace\sprintf()` and `$obj->sprintf()` are not the global `sprintf()` function, so the dynamic placeholder pattern should not apply.

### Actual behavior:

No errors or warnings. The sniff treats both calls as global `sprintf()` and applies the dynamic placeholder pattern handling.

## Environment

| Question                 | Answer
| ------------------------ | -------
| PHP version              | 8.4.19
| PHP_CodeSniffer version  | 3.13.5
| WordPressCS version      | develop
| PHPCSUtils version       | 1.2.1
| PHPCSExtra version       | 1.5.0
| WordPressCS install type | git clone
| IDE (if relevant)        | _N/A_

## Tested Against `develop` Branch?
- [x] I have verified the issue still exists in the `develop` branch of WordPressCS.

**Comment by rodrigoprimo** [comment]
I noticed that the false negatives described in this issue don't behave the same on PHPCS 3.x and PHPCS 4.x.

**Namespaced `sprintf()` calls** (e.g. `MyNamespace\sprintf()`, `\MyNamespace\sprintf()`, `namespace\sprintf()`)

These are only a false negative on PHPCS 3.x. On PHPCS 4.x, the tokenization of namespaced names changed, so the sniff no longer mistakes them for the global `sprintf()` and already flags them.

There is one edge case: `namespace\sprintf()` in a file without a namespace declaration resolves to the global `sprintf()`, so flagging it on PHPCS 4.x is actually a false positive rather than a correct result.

**Method calls to `sprintf()`** (e.g. `$obj->sprintf()`, `MyClass::sprintf()`)

These are a false negative on both PHPCS 3.x and PHPCS 4.x.

Since this issue was not reported before, I'm inclined to think we can wait until support for PHPCS 3.x is dropped before fixing it.
